### DISCUSSION TITLE
Struggling to find a virtual/touch keyboard

### DISCUSSION THREAD

**Discussion by markjsmith** [lxqt/lxqt#2588] [Q&A]
HI, I'm new to DietPi and LXQt but find it works amazingly on a RPI zero 2 and touch screen. The only problem is i have to plug a keyboard in as i cannot find a virtual keyboard.
Can someone talk me through what i need to do? I'm guessing i need to ensure a smooth autolog in first as i won't have a keyboard to select anything in shell!
Thanks

**Comment by stefonarch** [comment]
No idea about DietPi and software available but you could try one of those https://www.linuxlinks.com/best-free-open-source-virtual-keyboards/ (except the gnome one probably) and add it to autostart in Session settings.

**Comment by bbbhltz** [comment]
Looking through suggestions online and I see that many recommendations are out of date or no longer developed. That isn't bad, because lots of good software is also old.

http://t-sato.in.coocan.jp/xvkbd/ seems like an ok choice despite being a bit dated in appearance and it is in the repos https://packages.debian.org/bookworm/xvkbd

https://github.com/Alexander-r/qtvkbd is also pretty old but would likely work

https://packages.debian.org/bookworm/qtvirtualkeyboard-plugin is something to look at too

You should be able to add one of those to the Session settings and it will be there waiting

Auto login is probably a good idea.

First step though is to find a keyboard that you like and `apt install` it for example

```
$ su -
enter password
# apt install xbkdb
```

See if that works well with the touchscreen. If not, install something else and try it.

**Reply by Gnomasz** [reply]
I'm just going through trying out various on-screen keyboards and I can confirm that xvkbd "will do" - but I haven't tried using it on login screen.
If you want something with a future, I've seen that Corekeyboard is in active development, and Maliit was recently recommended to someone on the KDE forums, so it might get traction from there. Though I haven't gotten to trying any of them yet, though. I've seen some discouraging comments about Corekeyboard being unstable.

**Reply by markjsmith** [reply]
Thanks all. I've got xbkdb running for now but will look at the others as it's 'ok' but a bit limited as you say.
